openapis/swagger.json

```diff
@@ -1,594 +1,794 @@
+                }
+            }
         },
-        "host_authority": {
-          "description": "who is hosting this entity?",
-          "type": "array",
-          "items": {
-            "type": "string"
-          }
+        "AliasOutputInfo": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string"
+                },
+                "rev": {
+                    "type": "string",
+                    "pattern": "^[0-9a-f]{8}$"
+                },
+                "release": {
+                    "type": "string",
+                    "description": "has the entity been released?",
+                    "enum": [
+                        "public",
+                        "private",
+                        "controlled"
+                    ]
+                },
+                "size": {
+                    "type": "integer",
+                    "format": "int64",
+                    "description": "size in bytes"
+                },
+                "keeper_authority": {
+                    "description": "who is keeping this entry up to date?",
+                    "type": "string"
+                },
+                "host_authority": {
+                    "description": "who is hosting this entity?",
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "hashes": {
+                    "$ref": "#/definitions/HashInfo"
+                },
+                "metadata": {
+                    "type": "string"
+                }
+            }
         },
-        "hashes": {
-          "$ref": "#/definitions/HashInfo"
+        "VersionInputInfo": {
+            "type": "object",
+            "properties": {
+                "target": {
+                    "type": "string",
+                    "description": "how the version entries were collected",
+                    "enum": [
+                        "all",
+                        "latest"
+                    ]
+                }
+            }
         },
-        "metadata": {
-          "type": "string"
-        }
-      }
-    },
-    "AliasRef": {
-      "type": "object",
-      "properties": {
-        "name": {
-          "type": "string"
+        "VersionOutputRef": {
+            "type": "object",
+            "properties": {
+                "did": {
+                    "type": "string",
+                    "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"
+                },
+                "baseid": {
+                    "type": "string",
+                    "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$"
+                }
+            }
         },
-        "rev": {
-          "type": "string",
-          "pattern": "^[0-9a-f]{8}$"
+        "VersionOutputInfo": {
```

Comment:
As @philloooo suggests - if you augment the existing submission endpoint with a baseid in InputInfo and OutputInfo, then you can remove these Version* specific types which are redefining the type of objects in the index